lib/gmo.rb

```diff
@@ -37,8 +37,10 @@ def api(path, args = {}, verb = "post", options = {}, &error_checking_block)
           }
           raise GMO::Payment::ServerError.new(result.body, error_detail)
         end
-        # Parse the body as Query string
-        response = Rack::Utils.parse_nested_query(result.body.to_s)
+        # Transform the body to Hash
+        # "ACS=1&ACSUrl=url" => { "ACS" => "1", ACSUrl => "url" }
+        key_values = result.body.to_s.split('&').map { |str| str.split('=', 2) }.flatten
+        response = Hash[*key_values]
```

Comment:
`Rack::Utils.parse_nested_query` works in most case. But when
there is a `+` in the response body, it converts `+` to empty string (which is wrong).

Reply:
Are you talking about this case?

```ruby
[16] pry(main)> Rack::Utils.parse_nested_query("a=1+3")
=> {"a"=>"1 3"}
```



In this case, I think URL decoding is necessary.
Also, URL decoding is not possible with this implementation.

```ruby
[19] pry(main)> { a: "1+3" }.to_query.split('&').map { |str| str.split('=', 2) }.flatten
=> ["a", "1%2B3"]
```

If you get this kind of response, it may be a problem on the GMO server.

Reply:
@jiikko Let me describe this with more details.

The issue I encounter is: 
```ruby
result = gmo.exec_tran(**args) 
# gmo response.body = "ACS=1&PaReq=eJxV+Ff"
result
# { "ACS" => "1", "PaReq" => "eJxV Ff" } <= actual result, wrong
# expected result { "ACS" => "1", "PaReq" => "eJxV+Ff" }
```
The problem caused by 
```ruby
[147] pry(main)> Rack::Utils.parse_nested_query("ACS=1&PaReq=eJxV+Ff")
=> {"ACS"=>"1", "PaReq"=>"eJxV Ff"} #  Rack::Utils.parse_nested_query convert "+" to " " (empty string)
```
So I fix the way to parse the response. 
Does here need URL decoding? It's a response, not a query from a URL.
I applied this change to our product and it works.

Reply:
Thank you for your explanation!!! :bow:
I understood perfectly.